Consolidate Python Messages to use SerializedPyObj

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34497 Don't run user function until all UserRRefs in the args are confirmed
* #34496 Split deserialize from runPythonUdf and remove generatePythonUDFResult
* #34495 Use SerializedPyObj in PythonRpcHandler::generatePythonUDFResult
* #34494 Split deserialize from _run_function in RPC internal.py
* #34493 Use SerializedPyObj in PythonRpcHandler
* #34492 Remove _load_return_value from RPC internal.py
* **#34491 Consolidate Python Messages to use SerializedPyObj**
* #34490 Avoid copy contents in SerializedPyObj



Differential Revision: [D20347467](https://our.internmc.facebook.com/intern/diff/D20347467)

mrshenli committed:
Consolidate Python Messages to use SerializedPyObj

[ghstack-poisoned]

mrshenli committed:
Update on "Consolidate Python Messages to use SerializedPyObj"




[ghstack-poisoned]

zhaojuanmao commented on torch/csrc/distributed/rpc/python_call.cpp:
```diff
@@ -6,29 +6,31 @@ namespace torch {
-      std::move(pickledPayload_),
-      std::move(tensors_),
+      std::move(payload),
+      std::move(std::move(serializedPyObj_.tensors_)),
```
nit: there are two std::move

zhaojuanmao commented on torch/csrc/distributed/rpc/python_resp.cpp:
```diff
@@ -6,27 +6,31 @@ namespace torch {
   return Message(
-      std::move(pickledPayload_), std::move(tensors_), MessageType::PYTHON_RET);
+      std::move(payload),
+      std::move(std::move(serializedPyObj_.tensors_)),
```
same as above

mrshenli committed:
Update on "Consolidate Python Messages to use SerializedPyObj"




[ghstack-poisoned]

mrshenli committed:
Update on "Consolidate Python Messages to use SerializedPyObj"




Differential Revision: [D20347467](https://our.internmc.facebook.com/intern/diff/D20347467)

[ghstack-poisoned]

mrshenli committed:
Update on "Consolidate Python Messages to use SerializedPyObj"




Differential Revision: [D20347467](https://our.internmc.facebook.com/intern/diff/D20347467)

[ghstack-poisoned]

xush6528 commented on torch/csrc/distributed/rpc/python_call.cpp:
```diff
@@ -6,29 +6,31 @@ namespace torch {
 
 Message PythonCall::toMessage() && {
+  auto payload = std::vector<char>(
+      serializedPyObj_.payload_.begin(), serializedPyObj_.payload_.end());
```
We are making a copy here? 
We should target this consolidation not introducing performance overhead, right?
Can we "move" from `serializedPyObj_ `? (seems need to first refactor SerializedPyObj to have vector<char> instead of string).
